offerzen/global/employers/employers@7.js

```diff
@@ -0,0 +1,1618 @@
+        code: '998',
+      },
+      {
+        id: 'vu',
+        text: 'Vanuatu',
+        code: '678',
+      },
+      {
+        id: 'va',
+        text: 'Vatican City',
+        code: '39',
+      },
+      {
+        id: 've',
+        text: 'Venezuela',
+        code: '58',
+      },
+      {
+        id: 'vn',
+        text: 'Vietnam',
+        code: '84',
+      },
+      {
+        id: 'wf',
+        text: 'Wallis and Futuna',
+        code: '681',
+      },
+      {
+        id: 'eh',
+        text: 'Western Sahara',
+        code: '212',
+      },
+      {
+        id: 'ye',
+        text: 'Yemen',
+        code: '967',
+      },
+      {
+        id: 'zm',
+        text: 'Zambia',
+        code: '260',
+      },
+      {
+        id: 'zw',
+        text: 'Zimbabwe',
+        code: '263',
+      },
+      {
+        id: 'ax',
+        text: 'Åland Islands',
+        code: '358',
+      },
+    ].map(({ id, text, code }) => ({
+      id,
+      text,
+      code,
+      search: normaliseCharacters(`${code} ${text}`),
+    }));
+
+    // Search Functionality
+    function matchCustom(params, data) {
+      if (!params.term || params.term === '') {
+        return data;
+      }
+
+      const term = normaliseCharacters(params.term)
+        .replace(/[^a-z0-9 ]/gi, '') // Ignores everything that isn't a number or letter
+        .trim();
+      const search = data.search;
+      let fuzzyTerm = term;
+
+      // TODO: If the search is not a number, use looser matching criteria. Refine to weight letters in the same word higher
+      // if (!term.match(/[0-9]/)) {
+      //   fuzzyTerm = term.split("").join(".*");
+      // }
+
+      if (search?.match(fuzzyTerm)) {
+        return data;
+      }
+      return null;
+    }
+
+    // Format Data for Dropdown
+    function formatData(data) {
+      if (!data.id) {
+        return data.text;
+      }
+      let countryName = data.text;
+      let countryCode = data.code;
+      let $data = $(
+        `<span class="country-text-container"><span class="country-code">+${countryCode} </span><span class="country-name">(${countryName})</span></span>`
+      );
+      return $data;
+    }
+
+    // Format Selected Country Code
+    function formatCode(data) {
+      if (!data.id) {
+        return data.text;
+      }
+      let countryCode = data.code;
+      let $data = $(`<span class="country-code">+${countryCode}</span>`);
+      return $data;
+    }
+
+    const countryCodeSelector = originalLeadForm.find('.js-phone-country-code');
+    const phoneNumberInput = originalLeadForm.find('.js-phone-number');
+    const countryCodeOverride = originalLeadForm
+      .find('.js-default-country-code')
+      .text()
+      .toLowerCase();
+
+    // Initialise Select2
+    countryCodeSelector.select2({
+      data: phoneNumberCountries,
+      templateResult: formatData,
+      templateSelection: formatCode,
+      matcher: matchCustom,
+      dropdownParent: $('.js-country-code-dropdown'),
+    });
```

Comment:
Please move this into a function  - you can call it initializePhoneNumberField, or setupPhoneNumberField or similar